flink/src/test/java/org/apache/iceberg/flink/TestFlinkSchemaUtil.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import org.apache.flink.table.api.DataTypes;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestFlinkSchemaUtil {
```

Comment:
Looks like this doesn't test:
* Field comments
* Struct types
* Maps or lists that contain structs
* Maps with non-string keys

Are all of those supported? If so, can you add tests?

Reply:
Done

Reply:
I don't see test cases for lists with required elements or lists of structs. Can you add those?

Reply:
> Why use the list's nullability for whether the elements in the list are nullable? 

I thought the `ListType.ofOptional` says that whether the list is allowed to be nullable. Checked the code, seems it means whether the element in list are allowed to be nullable.
```java
  public static class ListType extends NestedType {
    public static ListType ofOptional(int elementId, Type elementType) {
      Preconditions.checkNotNull(elementType, "Element type cannot be null");
      return new ListType(NestedField.optional(elementId, "element", elementType));
    }
    //....
```

Sorry for this noise, let me check this code again.